### ISSUE TITLE
vm throws mysterious exception on unknown uri schemes

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#23803]
This broken program produces a mysterious error:
```dart
export 'unknown_scheme:to/a/file.dart';
```

The error looks like:
```
The null object does not have a getter '_uri'.
```

Turns out that the URI scheme is unknown, so the vm ends up with a null URI.

(Note this is different than #21693, where the VM gets a valid URL whose target doesn't exist)

cc @iposva-google

**Comment by sgmitrovic** [comment]
Fixed in 940cc2dab8a8

**Comment by sethladd** [comment]
Thanks @sgmitrovic !
